### ISSUE TITLE
[Accessibility] Add scrolling and zooming support for graphics tablet pens as in Okular

### ISSUE THREAD

**Issue by ghost** [johnfactotum/foliate#1199]
I'm not sure if this is GTK or WebGTK or Foliate issue, since no other GTK software that I use provide scrolling or zooming for graphics tablet where that makes sense to have: Evince, GNOME Web, Loupe. The only exception is GIMP, where they implemented that you can press ctrl + space and move cursor up and down to zoom in/out.

**Is your feature request related to a problem? Please describe.**
As of now, there no way to scroll and zoom in books using graphics tablet's (any tablet: with or without built-in screen) pen. You can only use 3 dots menu for zooming and bottom bar to move between pages.

Similarly to touchpad, user should be able to scroll and zoom with tablet pen without relying on keyboard or mouse or additional menus.

**Describe the solution you'd like**
Many (especially cheap or "simple" ones) tablets doesn't include more then 1 button on pen (sometimes even 0), and sometimes 0 buttons on tablet itself. And because of that you can't properly zoom or scroll content without using keyboard or mouse or touchpad.

Okular takes advantage of that fact, so it has 2 features that cover this cases.
For scrolling, they implemented that use can scroll content by clicking somewhere on view and moving up-down as you would do on your smartphone (this method available if you use "Navigation" tool and works with both mouse and tablet pen, so you can test this yourself):
![okular_mouse_scrolling.webm](https://github.com/johnfactotum/foliate/assets/54743395/79ea0c4d-00cb-4669-9d0f-d98fd3ddfa58)

For zooming with tablet pens they implemented similar feature which seems to work exclusively for tablet pen. You need to press first button that you have on pen and move it up/down to zoom in/out:
[okular_tablet_scrolling_zooming.webm](https://github.com/johnfactotum/foliate/assets/54743395/15b9a528-f9f6-4edf-85f7-2484f3abce78)
(first part of video showing that I scroll up and down with pen, while second show that I zooming with pen. Okular hides cursor if it used by pen, so that's why you can't see it on recording).

**Describe alternatives you've considered**
As alternative, you can use 3 dots menu to zoom:
![зображення](https://github.com/johnfactotum/foliate/assets/54743395/01cc9290-8605-447d-a3da-787a8810e1e6)

And bottom bar to move between pages:
![зображення](https://github.com/johnfactotum/foliate/assets/54743395/91327d4c-f0c8-440e-9c14-47595918f8b1)

And while you can use this options to navigate, they very slow and require significantly more moving your arm around, while with Okular you can place arm on one place and scroll and zoom (this is especially true for tablets without built-in screens) without much effort.

**Comment by johnfactotum** [comment]
Duplicate of #380. I guess it isn't too difficult to implement now, at least for reflowable books, now that there's the `scrollBy()` method on the paginator. Still, a separate hand tool mode seems a bit clunky. Probably I'd prefer something like allowing dragging only on the margins (and maybe bring back "tap to turn pages" as well, but only on the margins). Another idea is to treat pens the same way as touch, i.e. the hand mode would essentially be always on. Still another idea would be to use the buttons on the pen to navigate.

**Comment by ghost** [comment]
> I guess it isn't too difficult to implement now, at least for reflowable books, now that there's the `scrollBy()` method on the paginator. Still, a separate hand tool mode seems a bit clunky.

Yep, and something like android-style would be good compromise. There you can pick and scroll without any tool mode in any program. Take ReadEra as example (which is actually very close in what Foliate can do, like quotes, translate selected text, vocabulary, library support, colors themes, etc).

> Probably I'd prefer something like allowing dragging only on the margins (and maybe bring back "tap to turn pages" as well, but only on the margins). 

That sound good, but maybe some visualization where exactly this margins or some "tutorial" that explains this might be required, since this is not most obvious feature. Even ReadEra do provide tutorial for some basic functions that it has to make sure that user know that this features exist, including "tap to margin to turn page".

> Another idea is to treat pens the same way as touch, i.e. the hand mode would essentially be always on. Still another idea would be to use the buttons on the pen to navigate.

That also good idea, since most of time you scroll and zoom, rather then selecting text. But relying on button to navigation doesn't sound good to me, because, as I said, it's not that uncommon for tablets to NOT have any button at all (be it on tablet itself of on pen) so relying on them is not best idea.
